packages/public-api/src/middleware/common.ts

```diff
@@ -1,11 +1,34 @@
-import { ValidationError } from 'joi';
+import { ZodError } from 'zod';
 import createCorsMiddleware from 'cors';
-import express, { Request, Response, NextFunction } from 'express';
+import { Request, Response, NextFunction } from 'express';
 import { NotFoundError, respondWithError } from './response';
 
-// TODO: add whitelist origins
-export const cors = createCorsMiddleware({ maxAge: 1800 });
-export const parseJson = express.json();
+const whitelist = {
```

Comment:
Let's rename it as `originWhiteList`.
Please also add `localhost` origin to the staging white list